coda/coda_mdstore/tests/test_views.py

```diff
@@ -591,14 +591,34 @@ def test_raises_http404_file_handle_is_falsy(self, rf):
         with pytest.raises(http.Http404):
             views.bagURLList(request, bag.name)
 
-    @pytest.mark.xfail(reason='Refactor is required.')
-    def test_output_text(self):
-        assert 0
+    def test_output_text(self, rf):
+        self.getFileHandle.return_value.url = 'https://coda/testurl'
+        self.getFileHandle.return_value.readline.side_effect = ['This is sample testing text', '']
+        bag = FullBagFactory.create()
+        request = rf.get('/')
+        response = views.bagURLList(request, bag.name)
+        assert b'https://coda/is sample testing text' in response.content
```

Comment:
Maybe we should change the string returned by readline to be something that more closely resembles the kind of string that would actually be returned, because on first glance, I don't understand why it is desirable for the first word to have been removed.

Reply:
I think this sort of applies to the next two added tests too. It might make these tests easier to read through if the returned and mocked values are constructed to make it clearer what the expected result should be.

Reply:
Comments can also help with making tests easier to read through and understand.

Reply:
I think the strings themselves are better now, but yes, some descriptive comments would go a long ways towards making this more understandable too. I think the focus of these comments should not be on describing the lines of code themselves, but rather on why things are happening. What is special about the input in one test vs another? What makes one different? Why is the resulting value correct? Comments that answer those questions can be really helpful for tests that aren't showing something obvious.

Reply:
After discussing about what the tests are doing with @somexpert , I'm renaming the test names and modifying the doc strings for readability.